nixos/modules/services/networking/openvpn3.nix

```diff
@@ -0,0 +1,34 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.openvpn3;
+in
+{
+  ###### interface
+
+  options.services.openvpn3.client = {
+    enable = mkEnableOption "openvpn3 client. It might override your DNS servers; if that causes you problems, look into \"--dns-setup-disabled true\" or \"--dns-fallback-google true\" options in \"openvpn3 config-manage\" command";
```

Comment:
I think the general workflow for OpenVPN consists of getting your hands on a config file, which you'd then `openvpn3 config-import`, and then use it. `openvpn3 config-manage` isn't a replacement for a config file: you still need to have a config file, and `openvpn3 config-manage` just lets you override some common options.

As I understand it, an OpenVPN3 config file should just work out of the box, and everything you'd have to manage by yourself would be command-line flags. In other words, I don't think there's much to manage with NixOS.

About a startup option: that sounds pretty nice. Apparently, there's already an official solution for that: https://openvpn.net/blog/openvpn-3-linux-and-auth-user-pass/
I guess the question remains if it should be done now or can be done at a later stage. I personally think it's fine to add it a bit later down the line, perhaps in a separate PR.

Reply:
> I guess the question remains if it should be done now or can be done at a later stage. I personally think it's fine to add it a bit later down the line, perhaps in a separate PR.

I would prefer to do it in iterations. Also, I need non-permanent VPN connection, because it overrides some public sites that I want to access when not working on a project, that requires this VPN. So, NixOS options for this VPN would be useless for me. Don't know about other use cases.

Reply:
> I guess the question remains if it should be done now or can be done at a later stage. I personally think it's fine to add it a bit later down the line, perhaps in a separate PR.

Up to you. Maybe it could be a home-manager module since it's `~/.openvpn3` and not `/etc/openvpn3`.

Reply:
> I would prefer to do it in iterations. 

Yeah, likewise.
An interesting consideration, though, is that we already have `services.openvpn.servers`. I think it would be better to allow it to be used with `openvpn3-linux` backend instead of creating a new service from scratch. I will be looking into how that goes.
> So, NixOS options for this VPN would be useless for me.

Same here. My VPN connection uses a one-time authentication code as a password, so I can't really autostart it anyway.
> Up to you. Maybe it could be a home-manager module since it's `~/.openvpn3` and not `/etc/openvpn3`.

Yes, I think it would be best. It really is a user-scoped feature, so in HM it goes.

Reply:
> An interesting consideration, though, is that we already have `services.openvpn.servers`. I think it would be better to allow it to be used with `openvpn3-linux` backend instead of creating a new service from scratch. I will be looking into how that goes.

that sounds like a good idea.



> Yes, I think it would be best. It really is a user-scoped feature, so in HM it goes.

we can't rely on home-manager to properly implement services.

Reply:
> >   Yes, I think it would be best. It really is a user-scoped feature, so in HM it goes.
>
> we can't rely on home-manager to properly implement services.

That home-manager module would be a future addon. We aren't really relying on it, it's just a nice-to-have for more declarative configuration that won't work for everyone. (see above)